File: materials/orders-kotlin/app/src/main/kotlin/Staff.kt
// =========================================================
// Файл: Staff.kt
// Описание: Система управления персоналом склада.
// =========================================================

interface WarehouseWorker {
    fun processOrder()
    fun attendMeeting()
    fun getRest()
    fun swingingTheLead()
}

// HumanManager - Человек
class HumanManager : WarehouseWorker {

    override fun processOrder() {
        println("Manager is processing logic...")
    }

    override fun attendMeeting() {
        println("Manager is boring at the meeting...")
    }

    override fun getRest() {
        println("Manager is taking a break...")
    }

    override fun swingingTheLead() {
        println("Manager is watching reels...")
    }
}

// RobotPacker - Робот
class RobotPacker(val model: String) : WarehouseWorker {

    override fun processOrder() {
        println("Robot $model is packing boxes...")
    }

    override fun getRest() {
        println("Robot was taken for maintenance")
    }

    override fun attendMeeting() {
        println("ERROR: Robot cannot attend meetings")
    }

    override fun swingingTheLead() {
        error("CRITICAL ERROR: Robot cannot waste our money (we hope so)")
    }
}

// manageWarehouse - функция, которая работает со списком работников
fun manageWarehouse(workers: List<WarehouseWorker>) {
    println("\n--- Warehouse Shift Started ---")
    for (worker in workers) {
        worker.processOrder()
        worker.attendMeeting()
        worker.getRest()
        worker.swingingTheLead()
    }
}

File: materials/orders-python/staff.py
# staff.py
# =========================================================
# Файл: staff.py
# Описание: Система управления персоналом склада.
# =========================================================

from abc import ABC, abstractmethod


class WarehouseWorker(ABC):
    @abstractmethod
    def process_order(self) -> None:
        pass

    @abstractmethod
    def attend_meeting(self) -> None:
        pass

    @abstractmethod
    def get_rest(self) -> None:
        pass

    @abstractmethod
    def swinging_the_lead(self) -> None:
        pass


class HumanManager(WarehouseWorker):
    def process_order(self) -> None:
        print("Manager is processing logic...")

    def attend_meeting(self) -> None:
        print("Manager is boring at the meeting...")

    def get_rest(self) -> None:
        print("Manager is taking a break...")

    def swinging_the_lead(self) -> None:
        print("Manager is watching reels...")


class RobotPacker(WarehouseWorker):
    def __init__(self, model: str):
        self.model = model

    def process_order(self) -> None:
        print(f"Robot {self.model} is packing boxes...")

    def attend_meeting(self) -> None:
        print("ERROR: Robot cannot attend meetings")

    def get_rest(self) -> None:
        print("Robot was taken for maintenance")

    def swinging_the_lead(self) -> None:
        raise RuntimeError("CRITICAL ERROR: Robot cannot waste our money (we hope so)")


def manage_warehouse(workers):
    print("\n--- Warehouse Shift Started ---")
    for worker in workers:
        worker.process_order()
        worker.attend_meeting()
        worker.get_rest()
        worker.swinging_the_lead()